MPC/contribution.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-await-in-loop */
+import childProcess from 'child_process';
+import fs from 'fs';
+
+import download from 'download';
+
+const { spawn } = childProcess;
+
+async function contribution(circuitName, random) {
+  return new Promise((resolve, reject) => {
+    const zokrates = spawn('/app/zokrates', [
+      'mpc',
+      'contribute',
+      '-i',
+      `./params/${circuitName}`,
+      '-o',
+      `./params/out/${circuitName}`,
+      '-e',
+      random,
+    ]);
+
+    zokrates.stderr.on('data', err => {
+      reject(new Error(`Setup failed: ${err}`));
+    });
+
+    zokrates.on('close', () => {
+      resolve();
+    });
+  });
+}
+
+const main = async () => {
+  try {
+    if (!process.env.RANDOM) throw new Error('Please provide source of randomness');
+    if (!process.env.NAME) throw new Error('What is... your name? What is... your quest?');
+
```

Comment:
If you print the input randomness, you will see it does not match what you entered.